lib/src/working_copy.rs

```diff
@@ -734,22 +739,26 @@ impl TreeState {
         let matcher: Option<Box<dyn Matcher>> = match changed_files {
             None => None,
             Some(changed_files) => {
-                let repo_paths = changed_files
-                    .into_iter()
-                    .filter_map(|path| {
-                        match RepoPath::parse_fs_path(
-                            &self.working_copy_path,
-                            &self.working_copy_path,
-                            path,
-                        ) {
-                            Ok(repo_path) => Some(repo_path),
-                            Err(FsPathParseError::InputNotInRepo(_)) => None,
-                        }
-                    })
-                    .collect_vec();
+                let repo_paths = trace_span!("processing fsmonitor paths").in_scope(|| {
+                    changed_files
+                        .into_iter()
+                        .filter_map(|path| {
+                            match RepoPath::parse_fs_path(
+                                &self.working_copy_path,
+                                &self.working_copy_path,
+                                path,
+                            ) {
+                                Ok(repo_path) => Some(repo_path),
+                                Err(FsPathParseError::InputNotInRepo(_)) => None,
+                            }
+                        })
+                        .collect_vec()
+                });
 
-                let repo_path_set: HashSet<_> = repo_paths.iter().collect();
-                deleted_files.retain(|path| repo_path_set.contains(path));
+                trace_span!("retaining fsmonitor paths").in_scope(|| {
+                    let repo_path_set: HashSet<_> = repo_paths.iter().collect();
+                    deleted_files.retain(|path| repo_path_set.contains(path));
+                });
```

Comment:
This also seems like it should be fast. Does it take noticeable time?

Reply:
@martinvonz it takes around 10ms for the `deleting file states` and `retaining fsmonitor paths` steps in nixpkgs IIRC. Of course, there's no way to know that without having already instrumented it 😉. I think I was trying to account for all/nearly-all of the time spent in snapshotting, and there was a gap for these steps so I added instrumentation.

Reply:
Fair enough :) Feel free to leave it then.